src/lesson11/additionalTask/ArraySlava.java

```diff
@@ -0,0 +1,78 @@
+package lesson11.additionalTask;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class ArraySlava {
+    private int size;
+    private final int capacity;
+    private Integer[] elements;
+
+    public ArraySlava() {
+        capacity = 10;
+        size = 0;
+        elements = new Integer[capacity];
+    }
+
+    public ArraySlava(int capacity) {
+        this.capacity = capacity;
+        size = 0;
+        elements = new Integer[capacity];
+    }
+
+    public void add(int element) {
+        if (size == capacity) {
+            grow();
+        }
+        elements[size] = element;
+        size++;
+    }
+
+    public void remove(Integer elements) {
+        Integer[] newArray = new Integer[this.elements.length - 1];
+        int counter = 0;
+        for (Integer element : this.elements) {
+            if (Objects.equals(element, elements)) {
+                continue;
+            } else {
+                newArray[counter] = element;
+                counter++;
+            }
+            this.elements = newArray;
+        }
+        size--;
+
+    }
+
+    public void search(Integer elements) {
+        boolean flag = false;
+        for (Integer i : this.elements) {
+            if (Objects.equals(i, elements)) {
```

Comment:
I think, it make sense to change method parameter from Integer to int and use just ==